pkg/cache/external/scheduler_cache_dao.go

```diff
@@ -0,0 +1,94 @@
+ limitations under the License.
+*/
+
+package external
+
+import (
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type SchedulerCacheDao struct {
+	Statistics     SchedulerCacheStatisticsDao     `json:"statistics"`
+	Nodes          map[string]NodeDao              `json:"nodes"`
+	Pods           map[string]PodDao               `json:"pods"`
+	SchedulingPods map[string]PodSchedulingInfoDao `json:"schedulingState"`
+}
+
+type SchedulerCacheStatisticsDao struct {
+	Nodes                 int            `json:"nodes"`
+	Pods                  int            `json:"pods"`
+	Assumed               int            `json:"assumed"`
+	PendingAllocations    int            `json:"pendingAllocations"`
+	InProgressAllocations int            `json:"inProgressAllocations"`
+	PodsAssigned          int            `json:"podsAssigned"`
+	Phases                map[string]int `json:"phases"`
+}
+
+type NodeDao struct {
+	Name              string             `json:"name"`
+	UID               types.UID          `json:"uid"`
+	NodeInfo          v1.NodeSystemInfo  `json:"nodeInfo"`
+	CreationTimestamp time.Time          `json:"creationTimestamp"`
+	Annotations       map[string]string  `json:"annotations"`
+	Labels            map[string]string  `json:"labels"`
+	PodCIDRs          []string           `json:"podCIDRs"`
+	Taints            []v1.Taint         `json:"taints"`
+	Addresses         []v1.NodeAddress   `json:"addresses"`
+	Allocatable       v1.ResourceList    `json:"allocatable"`
+	Capacity          v1.ResourceList    `json:"capacity"`
+	Conditions        []v1.NodeCondition `json:"conditions"`
+}
+
+type PodDao struct {
+	Namespace         string               `json:"namespace"`
+	Name              string               `json:"name"`
+	GenerateName      string               `json:"generateName"`
+	UID               types.UID            `json:"uid"`
+	CreationTimestamp time.Time            `json:"creationTimestamp"`
+	Annotations       map[string]string    `json:"annotations"`
+	Labels            map[string]string    `json:"labels"`
+	NodeName          string               `json:"nodeName"`
+	Affinity          *v1.Affinity         `json:"affinity"`
+	NodeSelector      map[string]string    `json:"nodeSelector"`
+	PriorityClassName string               `json:"priorityClassName"`
+	Priority          *int32               `json:"priority"`
+	PreemptionPolicy  *v1.PreemptionPolicy `json:"preemptionPolicy"`
+	SchedulerName     string               `json:"schedulerName"`
+	Tolerations       []v1.Toleration      `json:"tolerations"`
+	Containers        []ContainerDao       `json:"containers"`
+	Status            v1.PodStatus         `json:"status"`
+}
+
+type ContainerDao struct {
+	Name      string                  `json:"name"`
+	Resources v1.ResourceRequirements `json:"resources"`
+}
+
+type PodSchedulingInfoDao struct {
+	Namespace       string    `json:"-"`
+	Name            string    `json:"-"`
```

Comment:
Is "-" intentional?

Reply:
Yes, we don't want these to emit via JS.